pytorch_lightning/callbacks/progress/tqdm_progress.py

```diff
@@ -135,6 +135,15 @@ def is_enabled(self) -> bool:
     def is_disabled(self) -> bool:
         return not self.is_enabled
 
+    @property
+    def _val_processed(self):
+        if self.trainer is None:
+            return 0
+        if self.trainer.state.fn == "fit":
```

Comment:
use `TrainerFn`?

Reply:
Since this is an "optional" callback and not an internal component, I prefer to use "fit" as a user would.

Reply:
Also, making the import will increase the chances of a circular import

Reply:
hmm. seeing that issue a lot lately regarding circular imports. I wonder if we can generalize it better somehow using some sort of trainer property.